bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.kt

```diff
@@ -173,13 +177,19 @@ internal class EventStore(
         when (delivery.deliver(payload, deliveryParams)) {
             DeliveryStatus.DELIVERED -> {
                 deleteStoredFiles(setOf(eventFile))
+                checkEventQueueEmpty()
                 logger.i("Deleting sent error file $eventFile.name")
             }
 
-            DeliveryStatus.UNDELIVERED -> undeliveredEventPayload(eventFile)
+            DeliveryStatus.UNDELIVERED -> {
+                undeliveredEventPayload(eventFile)
+                checkEventQueueEmpty()
+            }
+
             DeliveryStatus.FAILURE -> {
                 val exc: Exception = RuntimeException("Failed to deliver event payload")
                 handleEventFlushFailure(exc, eventFile)
+                checkEventQueueEmpty()
             }
```

Comment:
These can all be rolled into a single call, since it exists for each of the possible return values